plugins/woocommerce/src/Admin/Overrides/OrderTraits.php

```diff
@@ -80,4 +88,24 @@ public function get_item_shipping_tax_amount( $item ) {
 	public function get_item_coupon_amount( $item ) {
 		return floatval( $item->get_subtotal( 'edit' ) - $item->get_total( 'edit' ) );
 	}
+
+	/**
+	 * Calculate cart tax amount for line item/product.
+	 *
+	 * @param WC_Order_Item $item Line item from order.
+	 *
+	 * @return float|int
+	 */
+	public function get_item_cart_tax_amount( $item ) {
+		$order_taxes     = $this->get_taxes();
+		$tax_data        = $item->get_taxes();
+		$cart_tax_amount = 0;
+
+		foreach ( $order_taxes as $tax_item ) {
+			$tax_item_id      = $tax_item->get_rate_id();
+			$cart_tax_amount += isset( $tax_data['total'][ $tax_item_id ] ) ? (float) $tax_data['total'][ $tax_item_id ] : 0;
```

Comment:
Can you clarify a bit more about what's being calculated here? From what I can see it's only adding up the tax rates if those same ID's are found in the order taxes.

In what cases do the tax ID's not line up with tax ID's in the order? For example in which scenarios can we not add up just all the taxes recorded for this line item?

Reply:
It was actually copied from the original implementation in `Reports/Products/DataStore` here:
https://github.com/woocommerce/woocommerce/pull/53019/files#diff-d2438118f148f28d98ab96e1a12bdd753e462f4659970939f2fe3e2b87155f59L450-L456

And I think since an order line item would only have one tax rate (is it true?), we can just assign the rate to `$cart_tax_amount` when match. The commit is 40e9f29.

Reply:
With compound taxes, it can get complicated:
![image](https://github.com/user-attachments/assets/7c400451-4da7-46b3-8c43-4ed4a1443255)

As we can see now in the order we have both tax rates applied per line item (also for shipping since the tax rate is set for shipping):
![image](https://github.com/user-attachments/assets/8ca969a6-0fe6-4f9c-afe6-74da9bdf1c86)

It might not be common to configure this manually but there are automatic tax plugins that apply scenarios where they might have a state tax and a city tax. So we do have to take it into consideration.

Reply:
> It was actually copied from the original implementation in `Reports/Products/DataStore`

I can't think of a scenario where total order taxes doesn't include line item taxes, as generally we'd expect the order totals to be recalculated as soon as something changes. But I suppose since this code has been around for  along time we could leave it as is, as it still calculates the same thing.